### PR TITLE
Add organization sort

### PR DESCRIPTION
**Pull request by MohammadPCh**
- #118 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `ProjectResolver` for handling project-related queries.
	- Added sorting functionality for projects based on "vouch" or "flag" criteria.
	- New GraphQL object types: `OrganisationType`, `OrganisationProjectType`, `ProjectType`, and `ProjectsSortedByVouchOrFlagType`.
	- Added a function to extract selected fields from GraphQL queries.

- **Bug Fixes**
	- Improved error handling during database query execution.

- **Documentation**
	- Enhanced accessibility of `ProjectResolver` through exports for easier integration.

### CODE REVIEW

**Comment by MohammadPCh** [comment]
@aminlatifi can you review it? It's not finished but I really appreciate if you take a look and approve the way that I handle it. thanks

File: src/server-extension/resolver.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      const manager = await this.tx();
+
+      const vouchValue = sortBy === "vouch" ? true : false;
+
+      const selectedFields = getSelectedFields(info);
+      const fields = selectedFields.join(", ");
+
+      const query = `
+        SELECT 
+          ${fields},
```
Comment (aminlatifi) [inline src/server-extension/resolver.ts:30]: Note, you can only select fields either mentioned in the group by section or aggregated version of them

File: src/server-extension/resolver.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+          ON project.id = organisation_project.project_id 
+        LEFT JOIN organisation 
+          ON organisation_project.organisation_id = organisation.id 
+        WHERE 
+          organisation.id IN (${orgIds.map((orgId) => `'${orgId}'`).join(",")}) 
+          AND organisation_project.vouch = ${vouchValue}
+        GROUP BY 
+          project.id, 
+          organisation_project.id, 
+          organisation.id
```
Comment (aminlatifi) [inline src/server-extension/resolver.ts:44]: only `project.id` is not enough?!

File: src/server-extension/resolver.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Arg("offset", () => Number, { nullable: true }) offset: number = 0,
+    @Info() info: GraphQLResolveInfo
+  ): Promise<ProjectType[]> {
+    try {
+      const manager = await this.tx();
+
+      const vouchValue = sortBy === "vouch" ? true : false;
+
+      const selectedFields = getSelectedFields(info);
+      const fields = selectedFields.join(", ");
```
[the remark this answers is not in the document]
Comment (aminlatifi) [reply src/server-extension/resolver.ts:26]: The AI is right, I think this query will return a list of projects augmented with some aggregated value (sum), so the user must only be allowed to pass fields in a limited list (project fields)

**Review by aminlatifi — CHANGES_REQUESTED** [review]

**Comment by MohammadPCh** [comment]
@aminlatifi Thanks for your great suggestions!

I agree with you, and I'll limit it to just the projects' IDs. My initial goal was to make the resolver dynamic, but I realized it’s more efficient to retrieve only the projects' IDs from this endpoint and then make a separate request for the project details.

Thanks again for your input!

File: src/server-extension/resolver.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        GROUP BY 
+          project.id, 
+          organisation_project.id, 
+          organisation.id 
+        ORDER BY 
+          SUM(organisation_project.count) DESC
+          LIMIT ${limit} OFFSET ${offset};
+  `;
+
+      const rawProjects = await manager.query(query);
```
[the remark this answers is not in the document]
Comment (MohammadPCh) [reply src/server-extension/resolver.ts]: I think it's more optimized to use raw query, am I right?

[the remark this answers is not in the document]
Comment (MohammadPCh) [reply src/server-extension/resolver.ts]: What do you think @aminlatifi ?

[the remark this answers is not in the document]
Comment (MohammadPCh) [reply src/server-extension/resolver.ts]: @coderabbitai I think it's more optimized to use raw queries!

[the remark this answers is not in the document]
Comment (MohammadPCh) [reply src/server-extension/resolver.ts]: @coderabbitai ok, I changed it and now it's using Parameterized Query, Is it secure?

**Review by aminlatifi — APPROVED** [review]
Thanks @MohammadPCh

**Pull request merged**